azure/scope/machine.go

```diff
@@ -371,6 +382,51 @@ func (m *MachineScope) ProviderID() string {
 	return parsed.String()
 }
 
+// AvailabilitySet returns the availability set for this machine if available.
+func (m *MachineScope) AvailabilitySetSpec() (azure.ResourceSpecGetter, error) {
```

Comment:
@CecileRobertMichon What do you think about this function I added to get the spec? Unlike most of the spec getters, this could also return an error.

Reply:
I had a similar question for virtual machines spec. At the time, I ended up deciding not to return an error https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1861/files#diff-b6931ac61f37a13f66fc7836e039bf714a0553d5db4f313cc8ef31ca5f5647b9L156 because otherwise it might fail for Delete() as well, and I didn't want to block VM deletion on getting the VM sku from the cache, since delete should not be subject to validations. The other way to avoid this is to ignore the error for Delete but that's not ideal since ignoring an error could lead to bugs down the line. Instead, I think it's better to skip setting those values in the spec if the cache is nil, and in the Reconcile() / spec Parameters() function, return an error if required values are missing from the spec. Thoughts?

Reply:
That makes sense, I just pushed some changes. How does that look?